### PR TITLE
Update AbstractForm.php for Multilanguage fields

### LINKED ISSUE #28675: FO address fields validation not using mb_strlen for non-ASCII characters
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

Validation fails then using UTF-8 non-ASCII characters for City and Address fields for string length more than 32 chars for City and more than 64 for Address.
mbstring is installed
Tools::strlen works correct and return length using mb_strlen

Seems like some controller use regular php strlen, not mb_strlen or Tools::strlen

Admin forms works correct and accept full length fields.

### Expected behavior

Validation should pass for 64 characters in City and 128 characters in Address field

### Steps to reproduce

fill address form as customer with non-ASCII characters
32++ for City field
64++ for Address field

### PrestaShop version(s) where the bug happened

1.7.8.3

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by panariga**
use Tools::strlen instead of strlen for proper UTF-8 sting measurement

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.8.x
| Description?      | This function is used in 1.7.8.x to validate address fields which leads to shorten valid lenght than DB can store.
| Type?             | bug fix
| Category?         | FO
| BC breaks?        |  no
| Deprecations?     |  no
| Fixed ticket?     | Fixes #28675
| Related PRs       |
| How to test?      | 
| Possible impacts? | this function should not be used for fields that meant to be ASCII-only.

### CODE REVIEW

**Comment by ghost** [comment]
Hello @panariga can you update the description of your PR with the good branch `1.7.8.x` for label and change the branch you target.

**Comment by MatShir** [comment]
hi @panariga, we are planning a patch release for September. I would really like to see your contribution integrated, many user could benefit from your fix. Could you change of branch, like @okom3pom suggested you ? So we can start review and testing it :)

**Comment by panariga** [comment]
There was fail test because of unescaped apostrophe in description.
I already fix it.
But not sure why it doesnt requeue.
If you can't retest it, just make PR from your account. This will be faster
I think

**Comment by FabienPapet** [comment]
Hello @panariga and thank you for the fix. Can you please rebase your PR onto `1.7.8.x` branch so that we can test it ?

Thank you !

**Review by atomiix — APPROVED** [review]

**Review by kpodemski — APPROVED** [review]

**Comment by hibatallahAouadni** [comment]
@kpodemski why did you add the label `Waiting for PM`? do we need a PM approval before testing?

**Comment by kpodemski** [comment]
@hibatallahAouadni I think @MatShir has to decide if it's going to 1.7.8.8 or not

**Comment by eshraw** [comment]
@kpodemski I believe @MatShir wanted to include it in 1788 considering his previous comment.

**Review by sallemiines — COMMENTED** [review]
Hello @panariga 

Thanx for this PR ! it's LGTM QA :heavy_check_mark: , you can see the attached screen record 

[Untitled_ Nov 21, 2022 3_01 PM.webm](https://user-images.githubusercontent.com/31663369/203074380-5e010493-4cd9-437a-9cca-9b2b7c38c9b5.webm)

Thank you ! :rocket:

**Comment by ghost** [comment]
Thank you  @sallemiines && @panariga

**Pull request merged**
